docs/project-development.md

```diff
@@ -0,0 +1,189 @@
+build system. Changing one file will require `nix-build` to recompile
+the entire package. In order to get a dev environment where changing a
+module only rebuilds the affected modules, even across packages, a
+more incremental tool is required.
+
+`cabal-install` is the only tool that simultaneously supports Nix and
+GHCJS. The Nix expression in the project file uses `shells` to setup
+`nix-shell` sandboxes that Cabal can use to build your project. The
+`shells` field defines which platforms we'd like to develop for, and
+which packages' dependencies we want available in the development
+sandbox for that platform. Note that specifying `common` is important;
+otherwise it will be treated as a dependency that needs to be built by
+Nix for the sandbox. You can use these shells with `cabal.project`
+files to build all three packages in a shared incremental environment,
+for both GHC and GHCJS.
+
+```yaml
+-- cabal.project
+packages:
+  common/
+  backend/
+  frontend/
+```
+
+```yaml
+-- cabal-ghcjs.project
+compiler: ghcjs
+packages:
+  common/
+  frontend/
+```
+
+To build with GHC, use the `nix-shell` command to enter the sandbox
+shell and use `cabal` (which is supplied by the sandbox):
+
+```bash
+$ nix-shell -A shells.ghc
+[nix-shell:~/path]$ cabal new-build all
+```
+
+To build with GHCJS:
+
+```bash
+$ nix-shell -A shells.ghcjs
+[nix-shell:~/path]$ cabal --project-file=cabal-ghcjs.project --builddir=dist-ghcjs new-build all
+```
+
+You can also run commands in the nix-shell without entering it
+interactively using the `--run` mode. This is useful for scripting.
+
+```bash
+$ nix-shell -A shells.ghc --run "cabal new-build all"
+$ nix-shell -A shells.ghcjs --run "cabal --project-file=cabal-ghcjs.project --builddir=dist-ghcjs new-build all"
+```
+
+**Note:** Cabal may complain with `Warning: The package list for
+'hackage.haskell.org' does not exist. Run 'cabal update' to download
+it.` This can be ignored since we are using Nix instead of Cabal's
+solver. Nix uses a package snapshot similar to a Stackage LTS.
+
+Building frontends with GHC
+---
+
+GHCJS can be quite slow, especially if you are using Template
+Haskell. Building the frontend with GHC can drastically speed up build
+times, and enables you to test from GHCi for even faster
+reloads.
+
+JSaddle is a set of libraries that allows Reflex-DOM to swap out its
+JavaScript backend easily. By default, Reflex-DOM's `mainWidget` will
+work on GHC out of the box, using the `jsaddle-webkit2gtk` backend. So
+simply building your `frontend` package using GHC will produce a
+working native program that renders DOM using WebKit. This is
+recommended for native desktop releases.
+
+`jsaddle-warp` is an alternative JSaddle backend that uses a local
+`warp` server and WebSockets to control a browser from a native
+Haskell process. This is recommended to allow testing different
+browsers, and to make use of a browser's developer tools since the
+WebKit backend does not provide any.
```

Comment:
WebKit actually *does* have a full inspector hiding in there!  If you right-click, you should be able to get to it.  However, it isn't as nice to use as chrome/etc., so your overall point in this paragraph stands.